### PR TITLE
Update opentelemetry-proto to latest version.

### PR DESCRIPTION
**Pull request by dengliming**
closes [https://github.com/open-telemetry/opentelemetry-java/issues/1912](https://github.com/open-telemetry/opentelemetry-java/issues/1912)

### CODE REVIEW

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ static Status toStatusProto(SpanData.Status status) {
     // todo: Update this when the proto definitions are updated to include UNSET and ERROR
     Status.StatusCode protoStatusCode = Status.StatusCode.STATUS_CODE_OK;
     if (status.getCanonicalCode() == StatusCode.ERROR) {
-      protoStatusCode = Status.StatusCode.STATUS_CODE_UNKNOWN_ERROR;
+      protoStatusCode = Status.StatusCode.STATUS_CODE_ERROR;
```
Comment (jkwatson) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:175]: I think there's a little more we need to do here. See the comments in the proto:  https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/trace/v1/trace.proto#L239-L249

Comment (dengliming) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:175]: When `status.getCanonicalCode()` is `StatusCode.UNSET`, the `protoStatusCode ` should be set to `Status.StatusCode.STATUS_CODE_OK`. Is that right?

Comment (jkwatson) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:175]: I think we're also supposed to be setting the `deprecated_code` as well, if I read those comments correctly.

Comment (dengliming) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:177]: Can you help to confirm again? Thanks.

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java [hunk 2]
```diff
@@ -169,12 +171,16 @@ static Span toProtoSpan(SpanData spanData) {
   }
 
   static Status toStatusProto(SpanData.Status status) {
-    // todo: Update this when the proto definitions are updated to include UNSET and ERROR
     Status.StatusCode protoStatusCode = Status.StatusCode.STATUS_CODE_OK;
```
Comment (Oberon00) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:174]: I think there is a mistake here: For UNSET, we should use UNSET, not OK.

**Review by Oberon00 — CHANGES_REQUESTED** [review]

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java (hunk 2 above)
Comment (dengliming) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java]: Thanks. Can you review again?

**Review by Oberon00 — APPROVED** [review]

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java (hunk 2 above)
Comment (Oberon00) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java]: Thanks, looks good now!

File: exporters/otlp/src/test/java/io/opentelemetry/exporters/otlp/SpanAdapterTest.java [hunk 3]
```diff
@@ -109,17 +112,35 @@ void toProtoSpanKind() {
   }
 
   @Test
+  @SuppressWarnings("deprecation") // setDeprecatedCode is deprecated.
```
Comment (jkwatson) [inline exporters/otlp/src/test/java/io/opentelemetry/exporters/otlp/SpanAdapterTest.java:115]: this line of code makes me laugh, with the comment.  😆

**Review by jkwatson — APPROVED** [review]
Thanks! Looks good.

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java (hunk 1 above)
Comment (jkwatson) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java:179]: looks good!

**Pull request merged**